client/landing/gutenboarding/hooks/use-steps.ts

```diff
@@ -41,26 +41,22 @@ export default function useSteps(): Array< StepType > {
 	}
 
 	// Logic necessary to skip Domains or Plans steps
+	// General rule: if a step has been used already, don't remove it.
 	const { domain, hasUsedDomainsStep, hasUsedPlansStep } = useSelect( ( select ) =>
 		select( ONBOARD_STORE ).getState()
 	);
 	const plan = useSelect( ( select ) => select( ONBOARD_STORE ).getPlan() );
-	const hasPaidPlanFromPath = useHasPaidPlanFromPath();
+	const hasPlanFromPath = !! usePlanFromPath();
 
 	if ( domain && ! hasUsedDomainsStep ) {
 		steps = steps.filter( ( step ) => step !== Step.Domains );
 	}
 
-	// Don't show the mandatory Plans step:
-	// - if the user landed from a marketing page after selecting a paid plan (in this case, hide also the Features step)
-	// - if this is an Anchor.fm signup
-	// - if a plan has been selected using the PlansModal but only if there is no Features step
-	if (
-		hasPaidPlanFromPath ||
-		isAnchorFmSignup ||
```

Comment:
@mreishus I'm cleaning this check since the `Step.Plans` can't ever be in `steps`. See https://github.com/Automattic/wp-calypso/pull/48115/files#diff-adb74a6d0cb333fb8f930552304039e5016a1001286f3343a6907507fa087b99R21

Reply:
Sounds good.